enhancements/20240930-vxlan-enhancement.md

```diff
@@ -0,0 +1,197 @@
+The bonding interface will play as the VTEP which needs the IP address to be assigned automatically from DHCP.
+The `vxlan42` interface should also be created and attached to the brige interface like (2). 
+```
+on NODE_2
+ip link add vxlan42 type vxlan id 42 dstport 4789 local 192.168.5.22 dev cn-vxlan_1-bo
+ip link set vxlan42 master cn-vxlan_1_br
+
+on NODE_3
+ip link add vxlan42 type vxlan id 42 dstport 4789 local 192.168.5.33 dev cn-vxlan_1-bo
+ip link set vxlan42 master cn-vxlan_1-br
+```
+(4) Note that we don't use `group` option while creating the vxlan interface, the broadcast packets will not be sent to the multicast group to learn the FDB entry from other VTEPs.
+We need to manually add the FDB entry on each NODE so the VM on the VxLAN network can find each other by ARP. 
+```
+on Mgmt NODE:	add NODE_2, NODE_3 in the forwarding table. The broadcast packets would be correctly forwarded and the gratuitous ARP can work on top of it.
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.22
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.33
+bridge fdb append 02:22:22:22:22:22 dev vxlan42 dst 192.168.5.22
+bridge fdb append 02:33:33:33:33:33 dev vxlan42 dst 192.168.5.33
+
+on NODE_2:	add Mgmt NODE and NODE_3 in the forwarding table.
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.11
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.33
+bridge fdb append 02:11:11:11:11:11 dev vxlan42 dst 192.168.5.11
+bridge fdb append 02:33:33:33:33:33 dev vxlan42 dst 192.168.5.33
+
+on NODE_3:	add Mgmt NODE and NODE_2 
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.11
+bridge fdb append 00:00:00:00:00:00 dev vxlan42 dst 192.168.5.22
+bridge fdb append 02:11:11:11:11:11 dev vxlan42 dst 192.168.5.11
+bridge fdb append 02:22:22:22:22:22 dev vxlan42 dst 192.168.5.22
+```
+
+(5) The VxLAN interface on each NODE should play as the ARP proxy for VMs attached to the same VxLAN network.
+Then the packet destined to the VM have the ARP response from the corresponding VTEP and correctly forwarded.
+Adding the ARP entry for each VM for other NODEs which has no information for it. 
+```
+if the VM is on the Mgmt NODE, the following arp entry needs to be added on
+NODE_2/NODE_3:
+sudo ip neigh add 10.42.0.21 lladdr 02:11:11:11:11:11 dev vxlan0
+
+if the VM is on the NODE_2, then we need to add the ARP entry on
+Mgmt NODE/NODE_3:
+sudo ip neigh add 10.42.0.21 lladdr 02:22:22:22:22:22 dev vxlan0
+```
+
+(6) Use the VxLAN interface created in (3) to play as the router for all VMs attached to the same VxLAN network on the NODE.
+The `vxlan` interface can connect the external network via the uplink/bonding interface of the guest ClusterNetwork.
+Thus all VMs which route via the vxlan interface can connect to the external network.
+The `iptables` rules which do the masquerade should be added and the `default` route entry should also be configured on the attached VM.
```

Comment:
You should not add your GW ip address on vxlan interface if that interface is a bridge slave.

Another issue, do you consider having two VNI with the same IP range and internet access?
If so, your host route table will be confused if you have multiple overlap subnet.